plugin/core/windows.py

```diff
@@ -349,14 +349,10 @@ def window(self) -> sublime.Window:
         return self._window  # type: ignore
 
     def sessions(self, view: sublime.View, capability: Optional[str] = None) -> Generator[Session, None, None]:
-        file_name = view.file_name() or ''
-        scope = view2scope(view)
         for sessions in self._sessions.values():
             for session in sessions:
-                if session.config.match_document(scope):
-                    if capability is None or capability in session.capabilities:
-                        if session.state == ClientStates.READY and session.handles_path(file_name):
-                            yield session
+                if session.can_handle(view, capability):
```

Comment:
Could still be filtered in the `for` loop. :)
`for session in [session for session in sessions if session.can_handle(view, capability)]:`